stores/wallet.ts

```diff
@@ -88,14 +177,335 @@ export const useWalletStore = defineStore('wallet', () => {
+
+        // Prepare signature for registration
+        const message = JSON.stringify(
+          {
+            action: 'register',
+            evmWallet: walletAddress,
+            email: payload.email,
+            ts: Date.now()
+          },
+          null,
+          2
+        )
+
+        const signature = await signMessageAsync({ message })
+        try {
+          await getLikeCoinAPIFetch()('/users/new', {
+            method: 'POST',
+            body: {
+              from: walletAddress,
+              sign: signature,
+              payload: message,
+              platform: 'evmWallet',
+              user: payload.accountId,
+              email: payload.email,
+              magicUserId,
+              magicDIDToken
+            }
+          })
+        } catch (error) {
+          hasError = true
+          if (error instanceof FetchError) {
+            switch (error.data) {
+              case 'INVALID_USER_ID': {
+                toast.add({
+                  icon: 'i-heroicons-exclamation-circle',
+                  title: $t('account_register_error_invalid_account_id', { id: payload?.accountId }),
+                  timeout: 10000,
+                  color: 'red',
+                  ui: {
+                    title: 'text-red-400 dark:text-red-400'
+                  }
+                })
+                continue
+              }
+              case 'EMAIL_ALREADY_USED': {
+                toast.add({
+                  icon: 'i-heroicons-exclamation-circle',
+                  title: getEmailAlreadyUsedErrorData({
+                    email: payload?.email as string,
+                    walletAddress,
+                    boundEVMWallet: error.data?.evmWallet,
+                    boundLikeWallet: error.data?.likeWallet,
+                    loginMethod
+                  })?.data.description || error.data?.message,
+                  timeout: 10000,
+                  color: 'red',
+                  ui: {
+                    title: 'text-red-400 dark:text-red-400'
+                  }
+                })
+                continue
+              }
+              default:
+            }
+          }
+          throw error
+        }
+
+        // Fetch user info after registration
+        let userInfoRes: FetchLikerInfoResult | undefined
+        try {
+          userInfoRes = await useFetchLikerInfoByWallet(walletAddress, { nocache: true })
+        } catch (error) {
+          if (error instanceof FetchError && error.statusCode === 404) {
+            throw createError({
+              status: 401,
+              message: 'REGISTER_USER_NOT_FOUND'
+            })
+          }
+          throw error
+        }
+
+        if (!userInfoRes) {
+          throw createError({
+            status: 401,
+            message: 'CANNOT_FETCH_USER_INFO'
+          })
+        }
+
+        return true
+      } catch (error) {
+        hasError = true
+        throw error
+      }
+    } while (hasError)
+    return false
```

Comment:
Not sure if we should just throw error if register fails, instead of returning `false` but catching all error
I think it is better to make store function more functional and less UI logic, i.e. throw errors out, add toast /UI display from component/page/caller, instead of embedding `toast.add` into store